### PR TITLE
Add isNull/isNotNull operator

### PR DESCRIPTION
**Pull request by czpmango**
Add `IS NULL` and `IS NOT NULL`

Fixes [https://github.com/vesoft-inc/nebula-graph/issues/701](https://github.com/vesoft-inc/nebula-graph/issues/701)

### CODE REVIEW

File: src/common/expression/UnaryExpression.cpp [hunk 1]
```diff
@@ -75,6 +76,14 @@ const Value& UnaryExpression::eval(ExpressionContext& ctx) {
...
             break;
         }
+        case Kind::kIsNull: {
+            result_ = (operand_->eval(ctx)).isNull() ? true : false;
+            break;
+        }
+        case Kind::kIsNotNull: {
+            result_ = (operand_->eval(ctx)).isNull() ? false : true;
+            break;
+        }
```
Comment (CPWstatic) [inline src/common/expression/UnaryExpression.cpp:86]: These two ops should be kind of relational expression.

**Review by jievince — APPROVED** [review]

File: src/common/expression/UnaryExpression.cpp [hunk 2]
```diff
@@ -39,6 +39,7 @@ void UnaryExpression::resetFrom(Decoder& decoder) {
 
 
 const Value& UnaryExpression::eval(ExpressionContext& ctx) {
+    DCHECK(!!operand_);
    switch (kind_) {
```
Comment (laura-ding) [inline src/common/expression/UnaryExpression.cpp:43]: Adjust the indentation of the `switch`

File: src/common/expression/UnaryExpression.cpp (hunk 1 above)
Comment (laura-ding) [reply src/common/expression/UnaryExpression.cpp:86]: agree with @CPWstatic 
convert to `expression == NULL` or `expression != NULL`

Comment (laura-ding) [reply src/common/expression/UnaryExpression.cpp:86]: agree with @CPWstatic 
convert to `expression == NULL` or `expression != NULL`

Comment (czpmango) [reply src/common/expression/UnaryExpression.cpp:86]: > agree with @CPWstatic
> convert to `expression == NULL` or `expression != NULL`

For openCypher `expression == null` will always get null, should use `isNull` to judge null equality.

File: src/common/expression/UnaryExpression.cpp (hunk 2 above)
Comment (czpmango) [reply src/common/expression/UnaryExpression.cpp:43]: My mistake, thx~

File: src/common/expression/UnaryExpression.cpp [hunk 3]
```diff
@@ -99,6 +108,12 @@ std::string UnaryExpression::toString() const {
         case Kind::kUnaryDecr:
             op = "--";
             break;
+        case Kind::kIsNull:
+            op = "IS NULL";
```
Comment (laura-ding) [inline src/common/expression/UnaryExpression.cpp:112]: here maybe is " IS NULL", miss a space. you can add test for it

File: src/common/expression/UnaryExpression.cpp [hunk 4]
```diff
@@ -99,6 +108,12 @@ std::string UnaryExpression::toString() const {
...
+            break;
+        case Kind::kIsNotNull:
+            op = "IS NOT NULL";
```
Comment (laura-ding) [inline src/common/expression/UnaryExpression.cpp:115]: ditto

File: src/common/expression/UnaryExpression.cpp (hunk 3 above)
Comment (czpmango) [reply src/common/expression/UnaryExpression.cpp:112]: Already add test case.  
`2 IS NULL` or `IS NULL(2)`, Which way do you think is better?

Comment (laura-ding) [reply src/common/expression/UnaryExpression.cpp:112]: Consistent with the grammar is ok

Comment (czpmango) [reply src/common/expression/UnaryExpression.cpp:112]: ok.

**Review by Aiee — APPROVED** [review]

**Review by jievince — APPROVED** [review]

**Pull request merged**